eth/handler.go

```diff
@@ -45,6 +45,10 @@ const (
 	// txChanSize is the size of channel listening to NewTxsEvent.
 	// The number is referenced from the size of tx pool.
 	txChanSize = 4096
+	// txMaxBroadcastSize is the max size of a transaction that will be broadcasted.
+	// All transactions with a higher size will be announced and need to be fetched
+	// by the peer.
+	txMaxBroadcastSize = 4096
```

Comment:
Any particular reason for setting is as 4K? Or just an arbitrary number.

Lots of deployment transaction will exceed this threshold though.

Reply:
TBH, I would completely remove the notion of transaction broadcasts altogether. With all the MEV infrastructure in place currently, there's not much value in trying to be 0.5ms faster at propagating a transaction.

As for deploys, probably nobody will want to front run them anyway so those are even less time sensitive.

Reply:
The only catch is that this mechanism places quite a big initial burden on the node issuing the transaction, since all it's peers will want to retrieve it from this one place initially since nobody else has it. For that, perhaps we'd need a slightly smarter announcer that does it in stages, delaying announcing to everyone. 

E.g. Announce to sqrt(peers), wait 100-200ms, repeat. That way by the time we "fill" announce to all our peers, most of them will have announcements maybe from other peers too, thus not hammering only the source for the data.

Reply:
Yup, look goods to me.